src/components/run-button/run-button-hooks.js

```diff
@@ -0,0 +1,175 @@
+import { setRunButtonStatus } from 'redux/actions';
+
+function isWorthUpdate(
+    studyUpdatedForce,
+    fetcher,
+    lastUpdateRef,
+    nodeUuidRef,
+    nodeUuid,
+    invalidations
+) {
+    const headers = studyUpdatedForce?.eventData?.headers;
+    const updateType = headers?.[UPDATE_TYPE_HEADER];
+    const node = headers?.['node'];
+    const nodes = headers?.['nodes'];
+    if (nodeUuidRef.current !== nodeUuid) {
+        return true;
+    }
+    if (fetcher && lastUpdateRef.current?.fetcher !== fetcher) {
+        return true;
+    }
+    if (
+        studyUpdatedForce &&
+        lastUpdateRef.current?.studyUpdatedForce === studyUpdatedForce
+    ) {
+        return false;
+    }
+    if (!updateType) {
+        return false;
+    }
+    if (invalidations.indexOf(updateType) <= -1) {
+        return false;
+    }
+    if (node === undefined && nodes === undefined) {
+        return true;
+    }
+    if (node === nodeUuid || nodes?.indexOf(nodeUuid) !== -1) {
+        return true;
+    }
+
+    return false;
+}
+
+// this hook loads <runButtonType> state into redux, then keeps it updated according to notifications
+export function useRunButtonStatus(
+    studyUuid,
+    nodeUuid,
+    fetcher,
+    invalidations,
+    resultConversion,
+    runButtonType
+) {
+    const nodeUuidRef = useRef();
+    const studyUpdatedForce = useSelector((state) => state.studyUpdated);
+    const lastUpdateRef = useRef();
+    const dispatch = useDispatch();
+
+    const update = useCallback(() => {
+        nodeUuidRef.current = nodeUuid;
+        fetcher(studyUuid, nodeUuid)
+            .then((res) => {
+                if (nodeUuidRef.current === nodeUuid) {
+                    dispatch(
+                        setRunButtonStatus(
+                            runButtonType,
+                            resultConversion ? resultConversion(res) : res
+                        )
+                    );
+                }
+            })
+            .catch(() => {
+                dispatch(
+                    setRunButtonStatus(runButtonType, RunningStatus.FAILED)
+                );
+            });
+    }, [
+        runButtonType,
+        nodeUuid,
+        fetcher,
+        studyUuid,
+        resultConversion,
+        dispatch,
+    ]);
+
+    /* initial fetch and update */
+    useEffect(() => {
+        if (!studyUuid || !nodeUuid) {
+            return;
+        }
+
+        const isUpdateForUs = isWorthUpdate(
+            studyUpdatedForce,
+            fetcher,
+            lastUpdateRef,
+            nodeUuidRef,
+            nodeUuid,
+            invalidations
+        );
+        lastUpdateRef.current = { studyUpdatedForce, fetcher };
+        if (nodeUuidRef.current !== nodeUuid || isUpdateForUs) {
+            update();
+        }
+    }, [
+        update,
+        fetcher,
+        nodeUuid,
+        invalidations,
+        studyUpdatedForce,
+        studyUuid,
+    ]);
+}
+
+export function useNodeData(
```

Comment:
renamed hook to use-analysis-status and moved it to specific file
since useNodeData has been left unchanged, it's been left in study-container.js for now